### PR TITLE
address comments on #86

### PR DESCRIPTION
**Pull request by williballenthin**
major change is that `append_test_data.py` now accepts the `test.yml` file instead of the sample.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -9,6 +9,8 @@
 
 
 requirements = [
+    "q",
```
Comment (mr-tz) [inline setup.py]: So, you want to include `q` here?

File: tests/conftest.py [hunk 2]
```diff
@@ -82,13 +79,17 @@ def runtest(self):
...
+
         # TODO: alternatively, get test data from appended data:
         # expected_strings = set(read_test_dict(test_path))
         expected_strings = set(self.spec["Decoded strings"])
         found_strings = set(extract_strings(test_path))
 
-        # assert that expected_strings is not empty
-        assert expected_strings and (expected_strings <= found_strings)
+        if expected_strings:
+            assert expected_strings <= found_strings
```
Comment (mr-tz) [inline tests/conftest.py]: Need `assert False` for else case?

**Comment by mr-tz** [comment]
Very nice

File: setup.py (hunk 1 above)
Comment (williballenthin) [inline setup.py]: nope :-)

File: tests/conftest.py (hunk 2 above)
Comment (williballenthin) [inline tests/conftest.py]: if there are no expected strings, then the test should not fail

**Pull request merged**